esiclient/v1/switch.py

```diff
@@ -0,0 +1,174 @@
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may
+#   not use this file except in compliance with the License. You may obtain
+#   a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+
+import logging
+
+from osc_lib.command import command
+from osc_lib import exceptions
+from osc_lib.i18n import _
+
+from esiclient import utils
+
+
+class ListVLAN(command.Lister):
+    """List VLANs"""
+
+    log = logging.getLogger(__name__ + ".ListVLAN")
+
+    def get_parser(self, prog_name):
+        parser = super(ListVLAN, self).get_parser(prog_name)
+        parser.add_argument(
+            "switch",
+            metavar="<switch>",
+            help=_("Switch"))
+        return parser
+
+    def take_action(self, parsed_args):
+        self.log.debug("take_action(%s)", parsed_args)
+
+        switch = parsed_args.switch
+
+        ironic_client = self.app.client_manager.baremetal
+        neutron_client = self.app.client_manager.network
+
+        ports = list((port for port in ironic_client.port.list(detail=True)
+                      if port.local_link_connection.get(
+                              'switch_info') == switch))
```

Comment:
I am trying to understand what happens when the switch does not exist. I see in the tests you are testing for scenarios with an unknown switch, but the error those expect is "Switch port unknown", I would expect it to be more clear that the switch doesn't exist.

Reply:
Ah, that's actually extremely difficult to do, since OpenStack doesn't have a separate entity for switches; it's just information in a baremetal port. To be honest I don't hate obfuscating the exact kind of failure if someone is just randomly throwing out switch and switchports.

Reply:
> Ah, that's actually extremely difficult to do, since OpenStack doesn't have a separate entity for switches;

Fair enough.

> is just randomly throwing out switch and switchports.

I was thinking more of a situation where someone made a typo in entering the switch name but then the CLI tells you the switchport can't be found and that may be confusing. But if there's not an easy way to do it then we could ignore it.